### PR TITLE
component_monitoring.go: Remove use of NewRandomVMI func

### PR DESCRIPTION
**Pull request by Barakmor1**
NewRandomVMI is deprecated, libvmi should be used instead.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/monitoring/component_monitoring.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ package monitoring
 
 import (
 	"context"
+	"kubevirt.io/kubevirt/tests/libvmi"
```
Comment (fossedihelm) [inline tests/monitoring/component_monitoring.go:24]: please, move the import in the `kubevirt.io` section below

File: tests/monitoring/component_monitoring.go [hunk 2]
```diff
@@ -224,7 +225,7 @@ var _ = Describe("[Serial][sig-monitoring]Component Monitoring", Serial, decorat
 			err = virtClient.RbacV1().ClusterRoleBindings().Delete(context.Background(), "kubevirt-controller", metav1.DeleteOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			vmi := tests.NewRandomVMI()
+			vmi := libvmi.New()
```
Comment (fossedihelm) [inline tests/monitoring/component_monitoring.go:228]: There is another usage at line 243.
Also, `libvmi.New()` will produce a different vmi respect to `tests.NewRandomVMI()`.
The diff will be in:
- libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding())
- libvmi.WithNetwork(v1.DefaultPodNetwork())
- libvmi.WithResourceMemory(...)

Are those fields not required for this test?

**Review by fossedihelm — COMMENTED** [review]
Thank you @Barakmor1!
Please look at the comments.

File: tests/monitoring/component_monitoring.go (hunk 2 above)
Comment (dankenigsberg) [reply tests/monitoring/component_monitoring.go:228]: `libvmi.WithResourceMemory("1Mi")` is required for the VMI to start (without any guest). I don't think the network is necessary here.

(I got a reminder of the minimum memory on https://github.com/kubevirt/kubevirt/pull/11302 )

**Review by dankenigsberg — CHANGES_REQUESTED** [review]

File: tests/monitoring/component_monitoring.go (hunk 1 above)
Comment (Barakmor1) [reply tests/monitoring/component_monitoring.go]: Thanks :)

**Comment by fossedihelm** [comment]
/lgtm
Thank you!

**Review by dankenigsberg — APPROVED** [review]

**Comment by Barakmor1** [comment]
> @Barakmor1 Thanks for the PR! I would suggest to wait for #11374 to get merged and then use `NewGuestLess`

Sounds good to me :)

**Comment by enp0s3** [comment]
@Barakmor1 Sorry for the confusion, I see that #11401 was already merged, so you can rebasse and use `NewGuestless` already

**Review by dankenigsberg — APPROVED** [review]

**Review by enp0s3 — COMMENTED** [review]
/approve

**Pull request merged**